File: Ceil in BST.cpp
void dfs(Node* root,int v,vector<int>&lst){
    if(root->left)dfs(root->left,v,lst);
    int x=root->data;
    if(x>=v)lst.push_back(x);
    if(root->right)dfs(root->right,v,lst);
}
int findCeil(Node* root, int input){
    vector<int>lst;
    dfs(root,input,lst);
    if(lst.size()>0)return lst[0];
    return -1;
}

File: Ceil in BST.py

class Solution:
    def findCeil(self,root, inp):
        # code here
        self.lst=[]
        def dfs(root,inp):
            if root.data>=inp:self.lst.append(root.data)
            if root.left:dfs(root.left)
            if root.right:dfs(root.right)
        dfs(root,inp)
        self.lst.sort()
        return self.lst[0]
